src/Domain/Infra/Doctrine/MappingConfig.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MsgPhp\Domain\Infra\Doctrine;
+
+/**
+ * @author Pascal Wacker <hello@pascalwacker.ch>
+ *
+ * @internal
+ */
+final class MappingConfig implements MappingConfigInterface
+{
+    private $mappingFiles;
```

Comment:
lets' make this public and add a static `fromArray(array $config)` constructor, also we can write the implementation against known options (no need to make it over complex with _init and such)

i.e.

```php
public static function fromArray(array $config): self
{
  $instance = new self();
  $instance->mappingFiles = (array) $config['mapping_files'] ?? [];
  $instance->mappingDir = (string ) $config['mappign_dir'] ?? null;
  // etc.

  return $instance;
}
```